src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -19,255 +18,99 @@ namespace Microsoft.CodeAnalysis.PatternMatching
     /// Also, while the pattern matcher is culture aware, it uses the culture specified in the
     /// constructor.
     /// </summary>
-    internal sealed partial class PatternMatcher : IDisposable
+    internal abstract partial class PatternMatcher : IDisposable
     {
+        private static readonly char[] s_dotCharacterArray = { '.' };
+
         public const int NoBonus = 0;
         public const int CamelCaseContiguousBonus = 1;
         public const int CamelCaseMatchesFromStartBonus = 2;
         public const int CamelCaseMaxWeight = CamelCaseContiguousBonus + CamelCaseMatchesFromStartBonus;
 
-        private static readonly char[] s_dotCharacterArray = { '.' };
-
         private readonly object _gate = new object();
 
+        private readonly bool _includeMatchedSpans;
```

Comment:
This is what we decided in our last design meeting about this API.  We felt it was more appropriate as a facet of the matcher as users will tend to get the matcher once and use the same matcher for a bulk set of work.  Requiring them to pass these seem parameters to each match call was felt to be annoying.